src/Data/Replicator/Linear/Internal.hs

```diff
@@ -133,13 +133,26 @@ extend f = map f . duplicate
 --
 -- > elim (,) :: Replicator a %1 -> (a, a)
 -- > elim (,,) :: Replicator a %1 -> (a, a, a)
-elim :: forall (n :: Nat) a b f. (Elim (NatToPeano n) a b, IsFunN a b f, f ~ FunN (NatToPeano n) a b, n ~ Arity b f) => f %1 -> Replicator a %1 -> b
+elim ::
+  forall (n :: Nat) a b f.
+  ( -- Always fulfilled. Provides the actual implementation of 'elim' in
```

Comment:
Unfortunately, it seems that we can't add haddock comments on constraints :sad:

Reply:
What do you mean by “always fulfilled”?

Reply:
I mean: the end-user doesn't need to check something / implement something on their type to provide an instance of `Elim peanoN a b` (now that `f` is now longer part of the class parameters). There should always be an instance of `Elim peanoN a b` available for any `(peanoN, a, b)`.

My idea was to distinguish actual requirements (that is, the ones that are not fulfilled for any types, e.g. `Hash a` or `Ord a`) and constraints added just to help GHC with typechecking or to avoid recursion.

Reply:
I see. It makes sense, though it may not be the best way to say so. Basically, what you are trying to say is: “dear reader, please don't worry about these instances, they won't get in your way”. Be explicit about it.